src/tensor_product_basis.jl

```diff
@@ -0,0 +1,68 @@
+abstract type TensorProductBasis <: Function end
+
+##Chebyshev basis of the form [T_0(x), T_1(x), ..., T_n(x)] where T is the Chebyshev polynomial of the first kind
+struct ChebyshevBasis{Int} <: TensorProductBasis
+    n::Int
+    function ChebyshevBasis(n)
+        new{typeof(n)}(n)
+    end
+end
+
+function (basis::ChebyshevBasis)(x)
+    return [cos(j*acos(x)) for j in 1:basis.n]
+end
+
+##Fourier basis of the form F_j(x) = j is odd ? cos((j÷2)*x) : sin((j÷2)*x) => [F_0(x), F_1(x), ..., F_n(x)]
+struct FourierBasis{Int} <: TensorProductBasis
+    n::Int
+    function FourierBasis(n)
+        new{typeof(n)}(n)
+    end
+end
+
+function (basis::FourierBasis)(x)
+    return [sin(i*x) for i in 1:basis.n]
```

Comment:
this needs a constant term and cos terms in order to be a universal approximator.

Reply:
right now this can only approximate odd functions

Reply:
that said, a `sin` basis could be nice for odd functions